docs/zkapps/advanced-snarkyjs/time-locked-accounts.mdx

```diff
@@ -0,0 +1,114 @@
+
+As you can see, the maximum amount of unlocked tokens is defined by the `initialMinimumBalance`. It is called "initialMinimumBalance" because, even though the tokens show up in the balance, they can't be withdrawn — in other words, the account has a a non-zero _minimum balance_. Initially, that minimum balance is equal to the amount of tokens locked (= initial minimum balance). Over time, the minimum balance decreases until it hits zero, which is the condition that makes `isTimed` false again.
+
+Here's a breakdown of all timing fields:
+
+- **cliff time**: the initial time length during which no tokens whatsoever are unlocked.
+  - note: "time" is measured in slots (in Mina, currently 1 slot = 3 minutes).
+- **cliff amount**: the amount of tokens unlocked after the cliff time ended.
+  - if this is greater or equal the initial minimum balance, _all_ tokens get unlocked after the cliff time.
+- **vesting period**: after the cliff time, tokens get unlocked periodically in a fixed interval, by a fixed amount. The vesting period is the length of that interval.
+- **vesting increment**: the amount by which tokens get unlocked after each vesting period
+
+It's important to note that there is only ever one vesting schedule per account and it can't be changed during the vesting period. So, whenever `isTimed` is true, all the timing fields are immutable. When all tokens are unlocked and `isTimed` flips back to `false`, the account timing becomes mutable again.
+
+### Setting timing in SnarkyJS
+
+In SnarkyJS, `timing` is one of the account fields that can be updated by an account update:
+
+```ts
+accountUpdate.account.timing.set({ initialMinimumBalance, cliffTime, ...etc });
+```
+
+You need to set all timing parameters except `isTimed`, which is managed by the protocol automatically.
+
+Let's see how to correctly implement some simple example use cases.
+
+**Use case 1: all tokens unlocked after 1 week**. If you don't care about the linear vesting schedule and just want all tokens unlocked after a certain time, then the only parameters you really care about are `cliffTime` and `cliffAmount`. We also want to set `initialMinimumBalance` equal to `cliffAmount` to ensure all tokens are unlocked at this point. The values `vestingPeriod` and `vestingIncrement` do not matter so we set them to 1 and 0, respectively:
+
+```ts
+// example: 10 locked MINA
+const totalAmount = UInt64.from(10e9);
+
+// example: tokens locked for one week
+const cliffTime = UInt32.from((60 / 3) * 24 * 7);
+
+accountUpdate.account.timing.set({
+  initialMinimumBalance: totalAmount,
+  cliffTime,
+  cliffAmount: totalAmount,
+  vestingPeriod: UInt32.from(1), // 1 because an interval of length 0 is ill-defined behaviour
```

Comment:
Does SnarkyJS through an error if 0 is given? Feels like it should. Then would simplify this statement to say `// must be 1  or more; defaults to 1` or something

Reply:
turns out the protocol rejects 0 but snarkyjs by itself doesn't (except when testing with local blockchain)